mozjs/src/conversions.rs

```diff
@@ -536,7 +542,21 @@ pub unsafe fn latin1_to_string(cx: *mut JSContext, s: *mut JSString) -> String {
 
     let chars = slice::from_raw_parts(chars, length as usize);
     let mut s = String::with_capacity(length as usize);
-    s.extend(chars.iter().map(|&c| c as char));
+    let v = s.as_mut_vec();
+    for c in chars {
+        // This is safe as we are ensured that we only have "Latin1" strings which have the upper
+        // code word 00.
+        let i = std::mem::transmute::<_, AsciiChar>(*c).0 as u8;
+        v.push(i);
+    }
+
+    /* The above loop should be replaced with this which will allow simd optimization when ascii_char is stable
+    s.extend(
+        chars
+        .iter()
+        .map(|&c| std::ascii::Char::from_u8_unchecked(c)),
+    ),
+    */
```

Comment:
Please remove this commented out code. It's enough to say that above that the new `ascii::Char` should be used when stable.